man/systemd.network.xml

```diff
@@ -809,6 +809,15 @@
             <para>As in the <literal>[Network]</literal> section.</para>
           </listitem>
         </varlistentry>
+         <varlistentry>
+           <term><varname>GatewayOnlink=</varname></term>
+           <listitem>
+             <para>Specifies that the kernel does not have to check if the gateway is reachable directly
+             by the current machine (i.e. the kernel does not need to check if the gateway is attached to
+             the local network). Explicitly consider the gateway to be onlink, even if there is no route
```

Comment:
Sometimes, we  may want to send packets via a machine (gateway) which is physically connected to our network, but to which we don't specifically have a route in  kernel routing table.
In a situation like this, if we try to add a route through such a gateway, the kernel will perform consistency checks on the route we're trying to add and will detect that it doesn't know how to send packets to that gateway, rejecting the route. The "onlink" option tells the kernel that the it does not have to check if the gateway is reachable directly by the current machine (i.e., the kernel does not need to check if the gateway is attached to the local network), so that we can insert the route in the kernel table without it being complained about.